source/nodejs/adaptivecards-designer/src/designer-peers.ts

```diff
@@ -3317,4 +3413,146 @@ export class TablePeer extends TypedCardElementPeer<Adaptive.Table> {
             TablePeer.horizontalCellContentAlignmentProperty,
             TablePeer.verticalCellContentAlignmentProperty);
     }
-}
\ No newline at end of file
+}
+
+export class CarouselPeer extends ContainerPeer {
+
+    // Question: What do we want the default value to be here?
+    static readonly timerProperty = new CarouselTimerPropertyEditor(Adaptive.Versions.v1_6, "timer", "Timer", 5000);
+
+    protected internalAddCommands(context: DesignContext, commands: Array<PeerCommand>) {
+        super.internalAddCommands(context, commands);
+
+        commands.push(
+            new PeerCommand(
+                {
+                    name: "Add a page",
+                    isPromotable: false,
+                    execute: (command: PeerCommand, clickedElement: HTMLElement) => {
+                        let page = new Adaptive.CarouselPage();
+
+                        (this.cardElement as Adaptive.Carousel).addPage(page);
+
+                        this.updateChildren();
+                    }
+                }
+            )
+        )
+    }
+
+    initializeCardElement() {
+        super.initializeCardElement();
+
+        (this.cardElement as Adaptive.Carousel).addPage(new Adaptive.CarouselPage());
+    }
+
+    populatePropertySheet(propertySheet: PropertySheet, defaultCategory: string = PropertySheetCategory.DefaultCategory) {
+        super.populatePropertySheet(propertySheet, defaultCategory);
+
+        propertySheet.remove(CardElementPeer.isVisibleProperty,
+            ContainerPeer.bleedProperty, 
+            ContainerPeer.styleProperty);
+
+        propertySheet.add(
+            defaultCategory,
+            CarouselPeer.timerProperty);
+    }
+
+    canDrop(peer: DesignerPeer) {
+        return false;
+    }
+
+    isDraggable(): boolean {
+        return false;
+    }
+
+    attachOnPageChange() {
+        (this.cardElement as Adaptive.Carousel).onPageChanged = (activeIndex: number, realIndex: number) => {
+            const carouselElement = this.cardElement as Adaptive.Carousel;
```

Comment:
how do you use the onPageChanged?
why  do you update pages here?

Reply:
Update: plan is to disable loop mode while in the design mode, and handle the rerendering with the carousel event